client/src/Tools/_framework/Paths/AllAssignmentStudentData.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState } from "react";
+import { useLoaderData } from "react-router";
+
+import {
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  Link,
+  Tooltip
+} from "@chakra-ui/react";
+import axios from "axios";
+import { DoenetHeading as Heading } from "./Community";
+
+export async function action({ params, request }) {
+  return null;
+}
+
+export async function loader({ params }) {
+  const { data } = await axios.get(
+    `/api/getAllAssignmentStudentData/${params.userId}`,
+  );
+
+  let userId = Number(params.userId);
+
+  return {
+    userId,
+    studentScores: data.userScores,
+    assignments: data.assignments
+  };
+}
+
+export function AllAssignmentStudentData() {
+  const { userId, studentScores, assignments } =
+    useLoaderData();
+
+  useEffect(() => {
+    document.title = "Your Assignments";
+  });
+
+  const linkStyle = {
+    display: "block",
+    color: "var(--mainBlue)",
+  };
+
+  return (
+    <>
+        <Heading heading={"Your Assignments"} />
+
+        <Heading subheading="Score summary" />
+        <TableContainer>
+            <Table>
+            <Thead>
+                <Tr>
+                    <Th>Student</Th>
+                    {assignments.map((assignment) => {
+                        return (
+                            <Th key={`assignment${assignment.assignmentId}`}>
+                                <Link href={`/assignmentData/${assignment.assignmentId}`} style={linkStyle}>
+                                    {
+                                        assignment.name.length < 30 ? 
+                                        assignment.name
+                                        : <Tooltip label={assignment.name}>
```

Comment:
Nice! I've wanted this tooltip on our legacy gradebook for awhile, as I have some assignments that start the same and I can't tell them apart in the gradebook.